### PR TITLE
ENH: Refactored averaging/resampling/regression of data and added average properties to containers

### PR DESCRIPTION
**Pull request by pastewka**
This PR contains additions that are required to use `SurfaceTopography`'s container functionality in `TopoBank`. This is particular allows development and testing of whole-container analysis function on the `SurfaceTopgraphy` side and later deployment within `TopoBank`.

### CODE REVIEW

**Review by AntoineSIMTEK — CHANGES_REQUESTED** [review]
Looks good overall.

File: SurfaceTopography/Container/Averaging.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        lower, upper = np.min(x[x > 0]), np.max(x)
+        lower_decade, upper_decade = int(np.floor(np.log10(lower))), int(np.ceil(np.log10(upper)))
+
+        # Resample onto the same grid
+        collocation_points, bin_edges, resampled_y, variance = resample(
+            x, y, collocation='log', min_value=10 ** lower_decade, max_value=10 ** upper_decade,
+            nb_points=(upper_decade - lower_decade) * nb_points_per_decade + 1)
+
+        # Store for later averaging
+        for i, (xval, yval) in enumerate(zip(collocation_points, resampled_y)):
```
[the remark this answers is not in the document]
Comment (pastewka) [reply SurfaceTopography/Container/Averaging.py:85]: Oops, good catch. Will change this.

File: SurfaceTopography/Container/Averaging.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            results[lower_decade * nb_points_per_decade + i] += [(xval, yval)]
```
[the remark this answers is not in the document]
Comment (pastewka) [reply SurfaceTopography/Container/Averaging.py:86]: I tried that, but the problem is that floating-point rounding differences then give multiple entries that are essentially identical but appear under different keys.

[the remark this answers is not in the document]
Comment (AntoineSIMTEK) [reply SurfaceTopography/Container/Averaging.py:86]: Ah ok

File: SurfaceTopography/Nonuniform/VariableBandwidth.py [hunk 3]
```diff
@@ -100,21 +101,37 @@ def checkerboard_detrend_profile(line_scan, subdivisions, tol=1e-6):
...
-    line_scan : obj:`NonuniformLineScan`
+    self : obj:`NonuniformLineScan`
         Container storing the uniform topography map
+    quantities : str, optional
+        Specification of return tuple, where each string character
+        stand for specific quantity. Possible quantities are
+            - 'm': Magnification (Unit: dimensionless)
+            - 'b': Bandwidth (Unit: length)
+            - 'h': RMS height (Unit: length)
+            - 's': RMS slope (Unit: dimensionless)
```
[the remark this answers is not in the document]
Comment (pastewka) [reply SurfaceTopography/Nonuniform/VariableBandwidth.py:126]: You are right, the nonuniform does not return the slopes. The uniform have the option to do that. It is then the RMS of the detrending slope as in the arXiv.

**Review by AntoineSIMTEK — APPROVED** [review]

**Pull request merged**